fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/observer/EventImpl.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.fcrepo.kernel.impl.observer;
+
+import org.fcrepo.kernel.api.identifiers.FedoraId;
+import org.fcrepo.kernel.api.observer.Event;
+import org.fcrepo.kernel.api.observer.EventType;
+
+import java.net.URI;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.util.UUID.randomUUID;
+
+/**
+ * An event that describes one or more actions that a user preformed on a resource.
+ */
+public class EventImpl implements Event {
+
+    private final String eventId;
+    private final FedoraId fedoraId;
+    private final Set<EventType> types;
+    private final Set<String> resourceTypes;
+    private final String userID;
+    private final URI userURI;
+    private final Instant date;
+    // TODO I think this should go away and baseUrl and userAgent should be proper fields
+    private final Map<String, String> info;
```

Comment:
Just wondering why you didn't create proper fields give the above TODO?

Reply:
Because I didn't want to change the JMS code in this PR. The JMS code appears to more or less require baseURl (I think it will NPE without it). I'll do a second pass with changes to the Event interface and JMS code.